doc/development/plugins.rst

```diff
@@ -1,397 +1,398 @@
-      conversion between the two conventions takes place automatically
-      by the plugin device.
-
-* :func:`pennylane.Device.capabilities`: A class method which returns the dictionary of capabilities of a device. A
-  new device should override this method to retrieve the parent classes' capabilities dictionary, make a copy
-  and update and/or add capabilities before returning the copy.
-
-  Examples of capabilities are:
-
-  * ``'model'`` (*str*): either ``'qubit'`` or ``'cv'``.
+For example:
 
-  * ``'returns_state'`` (*bool*): ``True`` if the device returns the quantum state via ``dev.state``.
+.. code-block:: python
 
-  * ``'supports_inverse_operations'`` (*bool*): ``True`` if the device supports
-    applying the inverse of operations. Operations which should be inverted
-    have the property ``operation.inverse == True``.
+    class MyDevice(Device):
+        """My Documentation."""
 
-  *  ``'supports_tensor_observables'`` (*bool*): ``True`` if the device supports observables composed from tensor
-     products such as ``PauliZ(wires=0) @ PauliZ(wires=1)``.
+        def execute(self, circuits: QuantumScriptOrBatch, execution_config: "ExecutionConfig" = DefaultExecutionConfig):
+            return 0.0 if isinstance(circuits, qml.tape.QuantumScript) else tuple(0.0 for c in circuits)
 
-  *  ``'supports_tracker'`` (*bool*): ``True`` if it has a device tracker attribute and updates information with
-     it.
+    dev = MyDevice()
 
-  Some capabilities are queried by PennyLane core to make decisions on how to best run computations, others are used
-  by external apps built on top of the device ecosystem.
+    @qml.qnode(dev)
+    def circuit():
+        return qml.state()
 
-  To find out which capabilities are (possibly automatically) defined for your device ``dev = device('my.device')``,
-  check the output of ``dev.capabilities()``.
+    circuit()
 
-Adding arguments to your device
---------------------------------
+This execute method works in tandem with the optional ``Device.preprocss``, described below in more detail.
+Preprocessing turns generic circuits into ones supported by the device or errors out. Execution then
+turns those supported circuits into numerical results. 
 
-.. important::
+In a more minimal example, for any initial batch and config, we expect to be able to do:
 
-    PennyLane supports both qubit and continuous-variable (CV) devices. However, from
-    here onwards, we will demonstrate plugin development focusing on qubit-based devices
-    inheriting from the :class:`~.QubitDevice` class.
+.. code-block:: python
 
-Defining the ``__init__.py`` method of a custom device is not necessary; by default,
-the :class:`~.QubitDevice` initialization will be called, where the user can pass the
-following arguments:
+    transform_program, execution_config = dev.preprocess(initial_config)
+    batch, fn = transform_program(initial_batch)
+    fn(dev.execute(batch, execution_config))
```

Comment:
What is `fn` here?

Reply:
renaming to `postprocessing` and separately applying it into a different step.